### PR TITLE
Fix for Telegram backend mix backup locations when using multiple accounts

### PR DESCRIPTION
**Pull request by martikyan**
Telegram backend was using the account which was instantiated the first and it was using it for auth and the backup.

### CODE REVIEW

**Comment by warwickmm** [comment]
Thanks @martikyan.  I got the following error once, but I can't reproduce it anymore.  Have you encountered this before?
```
Command failed: Couldn't read the packet length
System.InvalidOperationException: Couldn't read the packet length
  at TLSharp.Core.Network.TcpTransport.Receive (System.Threading.CancellationToken token) [0x000e1] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at TLSharp.Core.Network.MtProtoSender.Receive (TeleSharp.TL.TLMethod request, System.Threading.CancellationToken token) [0x00088] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at TLSharp.Core.TelegramClient.RequestWithDcMigration (TeleSharp.TL.TLMethod request, System.Threading.CancellationToken token) [0x0016c] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at TLSharp.Core.TelegramClient.SendRequestAsync[T] (TeleSharp.TL.TLMethod methodToExecute, System.Threading.CancellationToken token) [0x00082] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at TLSharp.Core.TelegramClient.SendAuthenticatedRequestAsync[T] (TeleSharp.TL.TLMethod methodToExecute, System.Threading.CancellationToken token) [0x0009d] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at TLSharp.Core.TelegramClient.GetHistoryAsync (TeleSharp.TL.TLAbsInputPeer peer, System.Int32 offsetId, System.Int32 offsetDate, System.Int32 addOffset, System.Int32 limit, System.Int32 maxId, System.Int32 minId, System.Threading.CancellationToken token) [0x000e9] in <ff8f392d9e6f48368c05ac47637c2747>:0 
  at Duplicati.Library.Backend.Telegram.RetrieveMessages (TeleSharp.TL.TLInputPeerChannel inputPeerChannel, System.Collections.Generic.List`1[T] result, System.Int64 offsetDate) [0x00034] in <d0feef6775eb497c8d971e452d02c75f>:0 
  at Duplicati.Library.Backend.Telegram.ListChannelFileInfos () [0x0004d] in <d0feef6775eb497c8d971e452d02c75f>:0 
  at Duplicati.Library.Backend.Telegram+<>c__DisplayClass27_0.<Delete>b__0 () [0x0000d] in <d0feef6775eb497c8d971e452d02c75f>:0 
  at Duplicati.Library.Backend.Telegram.SafeExecute (System.Action action, System.String actionName) [0x00033] in <d0feef6775eb497c8d971e452d02c75f>:0 
  at Duplicati.Library.Backend.Telegram.Delete (System.String remotename) [0x00037] in <d0feef6775eb497c8d971e452d02c75f>:0 
```

File: Duplicati/Library/Backend/Telegram/Extensions/TelegramClientExtensions.cs [hunk 1]
```diff
@@ -18,31 +45,87 @@ public static bool IsReallyConnected(this TelegramClient client)
...
+        }
+
+        private static MtProtoSender GetMtProtoSender(this TelegramClient client)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            var senderFieldInfo = typeof(TelegramClient).GetField("sender", _bindingFlags);
```
Comment (warwickmm) [inline Duplicati/Library/Backend/Telegram/Extensions/TelegramClientExtensions.cs:125]: Is the use of reflection necessary?  I'm worried it makes us sensitive to internal refactorings of `TelegramClient`.

Comment (martikyan) [reply Duplicati/Library/Backend/Telegram/Extensions/TelegramClientExtensions.cs:125]: There was only one way to achieve the solution yet.
Later the reflection will be removed and the underlying library will be changed to [TgSharp](https://github.com/nblockchain/TgSharp)

Now they do not support 2-factor authentication so we can't use it yet.

**Comment by martikyan** [comment]
Dear @warwickmm 

The exception has nothing to do with the current PR and that is a bug in TLSharp (I guess). As I have researched a bit about the problem the only solution for this is to retry the WS call.

**Comment by warwickmm** [comment]
Thanks @martikyan.

**Comment by martikyan** [comment]
Thank you too @warwickmm !

**Pull request merged**